File: Question1Test.java
import org.junit.BeforeClass;
import org.junit.Test;
import static org.junit.Assert.*;

public class Question1Test {

  private static Question1 myUnit;
  private static String[] arr;

  // Processing the array
  @BeforeClass
  public static void setUp() throws Exception {
    myUnit = new Question1(15);  //Make array max 15 numbers
    arr = myUnit.getArr();
  }
  /*
  This one tests multiples of 3
   */
  @Test
  public void testMultiplesOfThree() {
    assertEquals(arr[2], "Boats"); //Multiples of 3
    assertEquals(arr[2], arr[5]); //Multiples of 3
  }
  /*
  This one tests multiples of 5
   */
  @Test
  public void testMultiplesOfFive() {
    assertEquals(arr[4], "Group"); //Multiples of 5
    assertEquals(arr[4], arr[9]); //Multiples of 5
  }

  /*
  This one tests multiples of 3 and 5
   */
  @Test
  public void testMultiplesOfThreeAndFive() {
    assertEquals(arr[14], "BoatsGroup"); //Multiples of 5
  }

}

File: question1_test.rb
require 'minitest/autorun'
require './question1'

class TestQuestion1 < Minitest::Test

  def setup
    @q = Question1.new(15)            #Make array max 15 numbers
  end

  def test_multiples_of_three
    assert_equal @q.arr[2], "Boats"   #Multiples of 3
    assert_equal @q.arr[2], @q.arr[5] #Multiples of 3
  end

  def test_multiples_of_five
    assert_equal @q.arr[4], "Group"   #Multiples of 5
    assert_equal @q.arr[4], @q.arr[9] #Multiples of 5
  end

  def test_multiples_of_three_and_five 
    assert_equal @q.arr[14], "BoatsGroup"  #Multiples of 3 and 5
  end
end